PR: Added new curriculum and replaced the old readme.

Drone944 opened the pull request:
The new curriculum has been added to the main readme and the old readme has been transferred to a folder named 'resources'.

ivinjabraham commented:
Can you squash your commits so that we don't litter the history?

Also, make the README.md be a brief introduction to the proposed contents of Vidyaratna (curriculum, year-wise checklist of tasks and additional resources for domains) and make the corresponding links.

Drone944 commented:
> Can you squash your commits so that we don't litter the history?
> 
> Also, make the README.md be a brief introduction to the proposed contents of Vidyaratna (curriculum, year-wise checklist of tasks and additional resources for domains) and make the corresponding links.

I have squashed the commits and changed the readme.
